pkg/kubelet/kubelet.go

```diff
@@ -1749,6 +1749,7 @@ func (kl *Kubelet) syncPod(pod *api.Pod, mirrorPod *api.Pod, podStatus *kubecont
 	// Kill pods we can't run.
 	if err := canRunPod(pod); err != nil || pod.DeletionTimestamp != nil {
 		if err := kl.killPod(pod, nil, podStatus); err != nil {
+			err = fmt.Errorf("error killing pod: %v", err)
```

Comment:
That doesnt change the error that a pod worker uses when it generates the event, I thought that was your goal...